src/mesh_tools.rs

```diff
@@ -0,0 +1,196 @@
+use std::collections::{HashMap, HashSet};
+
+use smallvec::SmallVec;
+
+use crate::convert::cast_usize;
+use crate::geometry::Geometry;
+use crate::mesh_topology_analysis::face_to_face_topology;
+
+/// Crawls the geometry to find continuous patches of geometry.
+/// Returns a vector of new separated geometries.
+#[allow(dead_code)]
+pub fn separate_isolated_meshes(geometry: &Geometry) -> Vec<Geometry> {
+    let face_to_face = face_to_face_topology(geometry);
+    let mut available_face_indices: HashSet<u32> = face_to_face.keys().cloned().collect();
```

Comment:
Rust nit: use `.copied()` instead of `.cloned()`. Does not make any runtime difference here, but expresses intent more clearly (`.copied()` is not available for items that are `!Copy`, but `.cloned()` is).